Added ability to embed the timer as a widget on an external website

LeonGeldsch opened the pull request:
I added a new script that allows the timer to be embedded on other websites. I also added a new HTML page and updated the README to give examples on how to integrate it.

Districtfine commented on dist/embed.js:
```diff
@@ -0,0 +1,356 @@
+
+function createEmbedElements() {
+  let content =
+    '<audio> <source src="media/alarm.mp3" /> <source src="media/ogg.mp3" /> </audio> <header> <h2> <u>20-20-20 Interval Timer</u> </h2> <div id="time-selector"> Every <input type="number" id="working_timeVal" name="working_timeVal" value="20" max="120" /> <select id="working_timeUnit" name="working_timeUnit"> <option value="minutes">minutes</option> <option value="seconds">seconds</option> </sel ...
```
See if you can add some formatting for this by using template literals:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Districtfine commented on src/scripts/embed.js:
Could you explain to me why this file is needed? It's nearly identical to dist/embed.js

Districtfine commented on embed-example.html:
Not needed, I think the explanation in the readme was sufficent

Districtfine commented on dist/embed.js:
```diff
@@ -0,0 +1,356 @@
+function start() {
+  createEmbedElements();
+
+  function selectElement(id, valueToSelect) {
```
Looks like from this point on the code is almost the same as scripts/index.js is there a way that we can have index.js reference code in here? Best to keep things as DRY as possible

Districtfine reviewed (changes requested):
Thanks for the PR! Mostly looks good to me, my biggest concern is the repeated content, would be good if there was a way to deduplicate

LeonGeldsch commented:
Thank you for the suggestions. I have simplified `embed.js` to import all of the scripts externally. I also reformatted the embedded HTML to be more readable.

LeonGeldsch commented:
I've also deleted the duplicate `embed.js` in the `dist` folder. The reason I created it in the first place is because I saw you were using babel to process your source files before uploading them. I tried to do the same (which is why I also updated the "build" command in the `package.json`) but for some reason I kept getting an error after parsing the code with babel, so I just changed it to copy.

I've deleted the duplicate file for now.

Districtfine commented:
Thanks! Looks good
